components/integrations_marketplace/marketplace_modal.jsx

```diff
@@ -0,0 +1,185 @@
+                    pluginsArray.map((p) => {
+                        return (
+                            <MarketplaceItem
+                                key={p.Manifest.id}
+                                id={p.Manifest.id}
+                                name={p.Manifest.name}
+                                description={p.Manifest.description}
+                                version={p.Manifest.version}
+                                isPrepackaged={false}
+                                downloadUrl={p.DownloadURL}
+                                homepageUrl={p.HomepageURL}
+                                itemState={p.State}
+                                onConfigure={this.close}
+                            />);
+                    }) : (
+                        <div className='no_plugins_div'>
+                            <br/>
+                            <PluginIcon className='icon__plugin'/>
+                            <br/>
+                            <br/>
+                            {noPluginsMessage}
+                            <br/>
+                            <br/>
+                            {installedList ? (
+                                <a onClick={() => this.changeTab(MarketplaceTabs.ALL_PLUGINS)}>
+                                    <FormattedMessage
+                                        id='marketplace_modal.install_plugins'
+                                        defaultMessage='Install Plugins'
+                                    />
+                                </a>
+                            ) : null
+                            }
+                        </div>
+                    )
+            }
+        </div>);
+    }
+
+    render() {
+        //WIP: To add pagination section
+        const input = (
+            <div className='filter-row filter-row--full'>
+                <div className='col-sm-12'>
+                    <QuickInput
+                        id='searchChannelsTextbox'
+                        ref='filter'
+                        className='form-control filter-textbox search_input'
+                        placeholder={{id: t('marketplace_modal.search'), defaultMessage: 'Search Plugins'}}
+                        inputComponent={LocalizedInput}
+                        onInput={this.doSearch}
+                    />
+                </div>
+            </div>
+        );
+
+        return (
+            <RootPortal>
+                <FullScreenModal
+                    show={Boolean(this.props.show)}
+                    onClose={this.close}
+                >
+                    <div className='MarketplaceModal'>
+                        <h1>
+                            <FormattedMessage
+                                id='marketplace_modal.title'
+                                defaultMessage='Plugins Marketplace'
+                            />
+                        </h1>
+                        {input}
+                        <Tabs
```

Comment:
Not a big fan of using react-bootstrap stuff for new things, but because we don't have yet a proper alternative to that, I think is the correct way to do it :)

Reply:
Possible ticket for creating our own Tab widget?

Reply:
Yes, that could be great. Probably the first version can use the bootstrap tab under the hood, and if we want to get rid of react-boostrap we only have to get rid of the component in the widget, not everywhere.

Reply:
@reflog you are working in the list of reusable components, makes sense to add the "Tabs/Tab" widgets?

Reply:
@jespino - yep, it will be part of the list!